### PR TITLE
refactor: s2 icon updates

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

Introducing the Spectrum 2 workflow icons! 🎉 

## Motivation and Context

With the rolling out of Spectrum 2's new designs, a new set of iconography has been created and with that, some improvements on how those icons are provided.  To adjust for this, breaking changes have been made to this repository's architecture in order to support providing iconography in it's new format:

**Spectrum CSS Workflow icons**

Found in `./dist/`, this folder contains the contents published to npm for `@adobe/spectrum-css-workflow-icons`. This package represents S2 iconography and will now ship 3 formats: components (i.e., web components), react, and svg. At the root of this package, you can find a manifest (`dist/manifest.json`) in JSON format of all available assets, as well as a prebuilt spritesheet for the SVGs called `assets/svg-spriteSheet/icons.svg`.

_Note_: Providing a PR preview link was out of scope for the work being done here but is planned for a future iteration.

### Setup

- Check out this repository locally
- `yarn install`
- To preview SVG icons: `yarn serve`

**TODO:** Need to find and summarize the changed icons and list out any deprecated and new icons.

## Screenshots

Screenshot 2024-05-02 at 9 39 04 AM

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by karstens** [comment]
@castastrophe I really like what you are doing, thank you so much! How can I test this without publishing anything? Just check out the branch and build it and manually look/check the `dist` folder?

@Hinschmi @jianliao Are you okay with this?

**Comment by jianliao** [comment]
@castastrophe ~~Do I need yarn 4 for this project?~~

Never mind. It is yarn v1, I figure it out from the `yarn.lock` file.

**Review by pfulton — APPROVED** [review]
Thanks for taking care of this huge effort. After getting Artifactory access configured correctly, I was able to follow the steps that you outlined in the description, ran the scripts, and verified that the icons looked like they should in the browser.

**Comment by jianliao** [comment]
LGTM

**Review by jianliao — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -57,11 +54,9 @@ The following tasks are available:
 | Command | Description | Examples |
 | --- | --- | --- |
 | `clean` | Cleans all output files in the dist directory | `yarn clean` |
-| `build` | Cleans the dist directory, copies website assets and then brings in the rsp, express, and workflow icons into dist | `yarn build` |
-| `build:icons` | Deletes and rebuilds the icon folder, can only be run by an Adobe employee with access | `yarn build:icons` |
-| `build:site` | Cleans the dist directory, copies website assets and prepares the dist output for the site preview | `yarn build:site` |
+| `build` | Deletes and rebuilds the icon folder, can only be run by an Adobe employee with access | `yarn build:icons` |
```
Comment (karstens) [inline README.md]: @castastrophe Little typo, I guess. The command at the end of the line is just `yarn build`.

**Review by karstens — APPROVED** [review]
Did not look at the changeset or git workflows, but the asset handling and tasks do look good to me. 👍

File: README.md (hunk 1 above)
Comment (castastrophe) [reply README.md]: Great catch thank you!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                  publish-dir: dist
+                  production-branch: main
+                  production-deploy: false
+                  netlify-config-path: ./netlify.toml
+                  github-token: ${{ secrets.GITHUB_TOKEN }}
+                  deploy-message: ${{ github.event.pull_request.title }}
+                  enable-pull-request-comment: true
+                  enable-commit-comment: false
+                  overwrites-pull-request-comment: true
+                  alias: pr-${{ github.event.number }}
```
Comment (castastrophe) [inline .github/workflows/build.yml]:
```suggestion
                  alias: deploy-preview-${{ github.event.number }}
```

**Pull request merged**